clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java

```diff
@@ -868,10 +875,14 @@ private void handleFailedRequest(boolean shouldWait) {
                  again. We may disconnect from the broker and connect to a broker that supports client
                  telemetry.
                 */
+                boolean shouldWait = isRetryable(maybeFatalException);
                 if (shouldWait) {
                     updateErrorResult(DEFAULT_PUSH_INTERVAL_MS, nowMs);
                 } else {
-                    log.warn("Received unrecoverable error from broker, disabling telemetry");
+                    boolean shouldLog = shouldLog(maybeFatalException);
```

Comment:
I would also just check `maybeFatalException instanceof UnsupportedVersionException` rather than hiding the check in a method. Do we really have to handle the case of the exception being wrapped by another exception?